BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -419,17 +425,22 @@ - (void)fetchItemsWithCompletion:(void (^)(NSArray *items))completion
 - (void)switchToEmptyStateWithError:(NSError *)error
 {
     // No error means it's just an empty folder.
+    NSString *errorMessage = [[NSString alloc] init];
     if (error == nil) {
-        ((UILabel *)self.tableView.backgroundView).text = NSLocalizedString(@"This folder is empty.", @"Label: Label displayed when the current folder is empty");
+        errorMessage = NSLocalizedString(@"This folder is empty.", @"Label: Label displayed when the current folder is empty");
     } else {
-        ((UILabel *)self.tableView.backgroundView).text = NSLocalizedString(@"Unable to load contents of folder.", @"Label: Label displayed when the current folder is empty");
+        errorMessage = NSLocalizedString(@"Unable to load contents of folder.", @"Label: Label displayed when the current folder is empty");
     }
+    self.folderInfoLabel.text = errorMessage;
+    self.tableView.backgroundView = self.folderInfoLabel;
```

Comment:
set the background view every time we switch to empty state?
Why not just keep the single setting of it in viewDidLoad?